Send Slack Reports

sienkin opened the pull request:
Updated script behaviour to now send Slack-based scan reports directly, rather than generating a template that is then subsequently used elsewhere. This functionality is now optional.

This change was made to resolve a logic issue where empty Slack messages could be generated and sent from a Concourse pipeline when a script error was encountered and before a scan report was generated. E.g. when a given ECR image and tag didn't exist - the script would error with an "Image not found" error but a slack message could still be triggered based on a non-existent JSON template.

Added new `SLACK_WEBHOOK` and `SLACK_CHANNEL` to supported vars and included as part of checking the environment
Replaced `generate_slack_json` function with new `send_slack_report` function that generates a fully formatted Slack webhook payload, rather than an attachment template
Updated README
Added `requests` to the pip requirements file

sienkin committed:
Updated script to now send slack reports itself, rather than generating a template file
Updated required environment variables to now include SLACK_* vars
Made sending of slack reports optional (enabled by default)

sienkin committed:
Added requests to requirements

sienkin committed:
Updated README for new vars and examples

Jata2k commented on README.md:
```diff
@@ -14,6 +14,9 @@ The script utilises the following environment variables.
 | AWS_REGION            | Yes      | -             | AWS region to connect to                                                    |
 | MIN_SEVERITY          | No       | `ALL`         | Sets the minimum vulnerability severity level to indicate a failure         |
 | MAX_WAIT              | No       | `120`         | Sets how long, in seconds, to wait for ECR scan results to become available |
+| SLACK_DISABLE         | No       | -             | If set and non-empty, will disable sending of Slack reports                 |
```
Is `SLACK_DISABLE` needed? Can we not infer this from the presence or absence of `SLACK_CHANNEL` and `SLACK_WEBHOOK`?

Jata2k commented on README.md:
```diff
@@ -14,6 +14,9 @@ The script utilises the following environment variables.
 | AWS_REGION            | Yes      | -             | AWS region to connect to                                                    |
 | MIN_SEVERITY          | No       | `ALL`         | Sets the minimum vulnerability severity level to indicate a failure         |
 | MAX_WAIT              | No       | `120`         | Sets how long, in seconds, to wait for ECR scan results to become available |
+| SLACK_DISABLE         | No       | -             | If set and non-empty, will disable sending of Slack reports                 |
```
Is `SLACK_DISABLE` needed? Can we not infer this from the presence or absence of `SLACK_CHANNEL` and `SLACK_WEBHOOK`?

sienkin replied:
@Jata2k Updated in [6c8f9b4](https://github.com/companieshouse/ci-ecr-scan-check/pull/2/commits/6c8f9b456a27cb6b89b81140355952e288b92105). Slack reporting is now based on whether `SLACK_CHANNEL` is set or not. This logic is used to determine whether `SLACK_WEBHOOK` is validated or not.

I chose this approach as, when used as a Concourse Task, `SLACK_WEBHOOK` will be defaulted

sienkin committed:
Removed SLACK_DISABLE var
Updated script logic to depend on whether SLACK_CHANNEL has been set
Updated Slack environment var validation to also depend on SLACK_CHANNEL
Updated README
